src/transformers/pipelines/image_text_to_text.py

```diff
@@ -0,0 +1,466 @@
+                `False` otherwise, but you can manually override that behaviour by setting this flag.
+
+        Return:
+            A list or a list of list of `dict`: Each result comes as a dictionary with the following key (cannot return a combination
+            of both `generated_text` and `generated_token_ids`):
+
+            - **generated_text** (`str`, present when `return_text=True`) -- The generated text.
+            - **generated_token_ids** (`torch.Tensor`, present when `return_tensors=True`) -- The token
+                ids of the generated text.
+            - **input_text** (`str`) -- The input text.
+        """
+        batch_size = kwargs.get("batch_size", 1)
+
+        if images is None and text is None:
+            raise ValueError("You must at least provide either text or images.")
+
+        if isinstance(text, (list, tuple, KeyDataset) if is_torch_available() else (list, tuple)) and isinstance(
+            text[0], (list, tuple, dict)
+        ):
+            # We have one or more prompts in list-of-dicts format, so this is chat mode
+
+            if isinstance(text[0], dict):
+                return super().__call__(Chat(text, images), **kwargs)
+            else:
+                if images is None:
+                    images = [None] * len(text)
+                chats = [Chat(chat, image) for chat, image in zip(text, images)]  # 🐈 🐈 🐈
+                return super().__call__(chats, **kwargs)
+
+        # If we are not in chat mode, we need both images and text
+        if images is None or text is None:
+            raise ValueError("You must provide both images and text when not using chat templates.")
+
+        if not isinstance(images, (list, tuple)):
+            images = [images]
+        if isinstance(text, str):
+            text = [text] * len(images)
+        if not isinstance(text[0], str):
+            raise ValueError("The pipeline does not support nested lists of prompts.")
+
+        if hasattr(self.processor, "image_token") and self.processor.image_token is not None:
+            image_token = self.processor.image_token
+            if isinstance(image_token, AddedToken):
+                image_token = image_token.content
+        else:
+            image_token = IMAGE_TOKEN
+        # Check number of image_token token in each text
+        num_images_in_text = [text_single.count(image_token) for text_single in text]
+        if sum(num_images_in_text) > 0:
+            if any(num > 1 for num in num_images_in_text) and batch_size > 1:
+                raise ValueError(
+                    "The pipeline does not support multiple images for a single prompt with batch_size > 1."
+                )
+            # Check if already nested images and consistency
+            if isinstance(images[0], (list, tuple)):
+                if len(images) != len(text):
+                    raise ValueError("The number of nested image groups and prompts should be the same.")
```

Comment:
afaik not all models require an image token to be in prompt. BLIP for ex has no requirement and Paligemma only suggests users to use 'image' token. Will this check work for such models?

Reply:
there's an initial check before this:
https://github.com/huggingface/transformers/blob/74375f7a5bf570835b54c607fedfdeb73f7fb28a/src/transformers/pipelines/image_text_to_text.py#L311-L325
so if there's no image tokens in any prompt, we don't enter this logic at all